### DISCUSSION TITLE
Connection pooling using more than 1 connection per thread

### DISCUSSION THREAD

**Discussion by farnoy** [jeremyevans/sequel#1922] [General]
In async Ruby, it's desirable to be able to use multiple connections in the same thread, for example:

```rb
require 'fiber_scheduler'

FiberScheduler do
  Fiber.schedule do
    DB.get{pg_sleep(1)}
  end
  Fiber.schedule do
    DB.get{pg_sleep(1)}
  end
end
```

This will take at least 2 seconds, because Sequel wants to execute both statements using the same connection. It may also cause race conditions because all fibers on the same thread will immediately obtain the "owned" connection and their execution & queries will be interleaved.

```rb
require 'fiber_scheduler'

FiberScheduler do
  Fiber.schedule do
    DB[].server(:primary).get{pg_sleep(1)}
  end
  Fiber.schedule do
    DB[].server(:replica).get{pg_sleep(1)}
  end
end
```

This scenario takes less than 2 seconds, because the queries get spread over separate connections, but this obviously isn't the intended use case for multiple servers.

I believe a configuration switch to disable the thread-local lookups would allow the same thread to utilize multiple connections - https://github.com/jeremyevans/sequel/blob/7a70ac6d719c21bcb404adf6159cbd2769d9246f/lib/sequel/connection_pool/threaded.rb#L87-L89

**Comment by HoneyryderChuck** [comment]
I think you're looking for https://sequel.jeremyevans.net/rdoc-plugins/files/lib/sequel/extensions/fiber_concurrency_rb.html

**Reply by farnoy** [reply]
That's exactly it, works perfectly. Thanks!
